apps/ejabberd/src/mod_mam_odbc_arch.erl

```diff
@@ -257,7 +257,8 @@ prepare_message(Host, MessID, UserID, LocJID = #jid{}, RemJID = #jid{lresource =
     SSrcJID = jid_to_stored_binary(Host, LocJID, SrcJID),
     SDir = encode_direction(Dir),
     Data = packet_to_stored_binary(Host, Packet),
-    TextBody = mod_mam_utils:packet_to_search_body(mod_mam, Host, Packet),
+    TextBody0 = mod_mam_utils:packet_to_search_body(mod_mam, Host, Packet),
+    TextBody = unicode:characters_to_binary(TextBody0),
```

Comment:
Similar comment as with escaping - I think it would be better to ensure that `packet_to_search_body` already returns a utf8 binary (where it would, in turn, require ensuring that `mod_mam_utils:normalize_search_text` returns utf8 binary).

Reply:
in this same PR we changed mod_mam_utils:normalize_search_text in order to return utf8 binaries